src/sql/store.py

```diff
@@ -177,5 +182,51 @@ def _flatten(elements):
     return [element for sub in elements for element in sub]
 
 
+def store_snippet_as_sql(sql_command, snippet_name):
+    """
+    Store snippet as a .sql file
+
+    Parameters
+    ----------
+    command : str
+        query to be saved as the snippet .
+
+    snippet_name : str
+        Name of the saved snippet
+    """
+
+    snippet_path = Path(SNIPPETS_DIR) / f"{snippet_name}.sql"
+    snippet_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(snippet_path, "w") as file:
+        file.write(sql_command)
+    message = """Manual editing of .sql files may not be reflected when
+    reopening the notebook. Please edit snippets directly in the notebook
+    to ensure consistency."""
+
+    display.message(message, style="font-size: 12px; font-style: italic;")
```

Comment:
this message is too verbose, it'd be better to display it only when the user sets `persist_snippets = True`, in the magic.py we already have an example of how to do it: you can configure an observer that can run some function whenever a config parameter changes its value. you can print this message when it's set to true.

instead, when saving a snippet, you can display: "saving {snippet_name} to {path_to_sql_file}"

Reply:
Ok, I see what you mean. you're restoring the state of the snippets, that makes sense. however. I encountered an issue. if I stored snippets in a first notebook, and then open a second notebook and try to use them, they'll break 


see second.ipynb here: [testing-persist-sql.zip](https://github.com/ploomber/jupysql/files/12186794/testing-persist-sql.zip)

Reply:
@edublancas so i looked at the issue, 
I think I know what is causing it, fixing this should resolve it 
> I noticed that if I created a snippet and then a second snippet that depends on the first one, the content stored for the first one does not include the `WITH` statement, but it should. essentially, the SQL file should contain the output of `%sqlcmd snippets {name}`
> 
> notebook: [testing-persist-sql.zip](https://github.com/ploomber/jupysql/files/12186654/testing-persist-sql.zip)

Reply:
Hey, this issue is solved now. 
We do not encounter any errors in the second notebook

Reply:
@edublancas, I have made the changes, 
can you please review?